packages/fxa-content-server/app/scripts/views/subscriptions_product_redirect.js

```diff
@@ -0,0 +1,29 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import FormView from './form';
+import Template from 'templates/subscriptions_redirect.mustache';
+import PaymentServer from '../lib/payment-server';
+
+class SubscriptionsProductRedirectView extends FormView {
+  mustAuth = true;
+  template = Template;
+
+  initialize (options) {
+    this._currentPage = options.currentPage;
```

Comment:
Is this endpoint for RPs to point to so that a user can manage a subscription? If so, I'm going to advocate a different approach, one that is more in line with the spirit of OAuth with the added benefit that it provides OAuth based security.

What I see happening is that the RP first bounces off of the subscription server. The subscription server sets up a state token, stores it in a session cookie, then bounces to FxA, possibly making use of `prompt=none`. If the user is signed in, we immediately bounce back to the subscription service w/ an OAuth code and the state token. The subscription service checks the state token is the same as the one that is stored in cookie, then uses the code to get an access token and profile info.  Mermaid diagram incoming.

Reply:
> Is this endpoint for RPs to point to so that a user can manage a subscription?

This is where a marketing-oriented product page links to let a user sign up & pay for a subscription. Other management functions happen from the FxA settings page.

>  If the user is signed in, we immediately bounce back to the subscription service w/ an OAuth code and the state token. The subscription service checks the state token is the same as the one that is stored in cookie, then uses the code to get an access token and profile info. Mermaid diagram incoming.

For the most part, we're assuming that this link comes from a static product details page. There will be no web-based RP server to perform any of this work. The first RP will be a desktop-based product that will sign in separately.

Reply:
I think what @shane-tomlinson is describing here is something we discussed earlier: making the payments server a relying party, and relying on OAuth redirect flows to handle this case. I believe the outcome of that discussion was that it seemed like a good idea, but we weren't sure about making that change at that time. At least from my perspective, it seems like we're kind of in the same place as we were before, and should keep moving forward on the current solution, even if we make the payments server an RP in the future.

Reply:
Let's figure this out in Canada, I'm still uncomfortable passing an OAuth token directly. I think with [prompt=none](https://github.com/mozilla/fxa/pull/1150) we have a great path forward for both this use case and RPs that want to link directly to a subscription management page.